core/src/main/scala/org/bitcoins/core/crypto/BIP39Seed.scala

```diff
@@ -0,0 +1,69 @@
+package org.bitcoins.core.crypto
+
+import javax.crypto.SecretKeyFactory
+import javax.crypto.spec.PBEKeySpec
+import org.bitcoins.core.protocol.NetworkElement
+import org.bitcoins.core.util.Factory
+import scodec.bits.ByteVector
+
+sealed abstract class BIP39Seed extends NetworkElement {
+
+  def toExtPrivateKey(keyVersion: ExtKeyVersion): ExtPrivateKey =
+    ExtPrivateKey.apply(keyVersion, Some(this))
+}
+
+/**
+  * @see [[https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki# BIP32]]
+  */
+object BIP39Seed extends Factory[BIP39Seed] {
+
+  /**
+    * Generates a [[https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki BIP32]]
+    * seed from a sequence of bytes. Must be between 16 and 64 bytes.
+    */
+  override def fromBytes(bytes: ByteVector): BIP39Seed =
+    BIP39SeedImpl(bytes)
+
+  private val MIN_SEED_LENGTH_BYTES = 16
+
+  private val MAX_SEED_LENGTH_BYTES = 64
+
+  private case class BIP39SeedImpl(bytes: ByteVector) extends BIP39Seed {
+    require(
+      bytes.length <= MAX_SEED_LENGTH_BYTES && bytes.length >= MIN_SEED_LENGTH_BYTES,
+      // $COVERAGE-OFF$
+      s"Seed must be between $MIN_SEED_LENGTH_BYTES and $MAX_SEED_LENGTH_BYTES bytes, got ${bytes.length}"
+      // $COVERAGE-ON$
+    )
+  }
+
+  private val PSEUDO_RANDOM_FUNCTION = "PBKDF2WithHmacSHA512"
+  private val ITERATION_COUNT = 2048
+  private val DERIVED_KEY_LENGTH = 512
+
+  private val secretKeyFactory =
+    SecretKeyFactory.getInstance(PSEUDO_RANDOM_FUNCTION)
+
+  /**
+    * Generates a [[https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki BIP32]]
+    * seed from a mnemonic code. An optional password can be supplied.
+    * @param password Defaults to the empty string
+    */
+  def fromMnemonic(mnemonic: MnemonicCode, password: String = ""): BIP39Seed = {
+    val salt = s"mnemonic$password"
+
+    val words = mnemonic.words.mkString(" ")
+
+    val keySpec = new PBEKeySpec(
```

Comment:
named arguments here please

Reply:
It's a java method, cannot have named arguments :-(